### PR TITLE
Fix data, lambda, and GitHub workflow configs

### PR DESCRIPTION
**Pull request by chuckwondo**
- Adjust collection definitions and test fixtures to support thumbnails
- Bump lambda function mem sizes to 3008MB to optimize speed and cost
- Bump configure-aws-credentials GitHub Actions to v2, which uses Node 16, and eliminates GitHub Actions warnings about using Node 14

### CODE REVIEW

File: app/stacks/cumulus/main.tf
Comment (krisstanton) [inline app/stacks/cumulus/main.tf:1]: This is great! 
Just want to add a comment here in context.

I think after we run this a few times, we should sample the results for each of the individual service to see if we need to individually tweak them slightly up or down.

Before this change, we had some functions that run at 128 MB and fully utilize it.

Two possible scenarios (for each service) I could see after changing it to 3008 MB are:
	(1) we have something under 3008 MB being utilized: 1234 MB utilized
		In this case, we would need to come back and dial down the memory usage a little bit to see if the utilization stays the same.
	(2) we have the 3008 MB maxed usage: 3008 MB utilized
		In this case, we would need to come back and dial it up some more. 
		
In summary, this might be an ongoing optimization that we would need to revisit at least a couple more times.

**Review by krisstanton — APPROVED** [review]
Approved

File: app/stacks/cumulus/main.tf
Comment (chuckwondo) [reply app/stacks/cumulus/main.tf:1]: What I was trying to convey on our recent call about this was that there is an article (which I'm still trying to locate) that describes lambda tuning that the author performed, which showed 3008MB as the sweet spot of performance and cost.

Specifically, even for lambda functions that require far less memory than 3008MB, because dialing up the memory also automatically dials up the vCPUs allocated, such lambda functions ran more quickly than they did with lower memory settings. Thus, the outcome was that the lambdas were cheaper to run, even though they were allocated more memory and vCPUs than required. The dramatic speed improvement outweighed the additional resource allocation, so the cost was reduced.

As we saw in the logs the other day, an extremely simply lambda function was taking 6 seconds to execute with memory set at 128MB (IIRC), while another, slightly more complex lambda function was taking 1/2 second with memory set at 3008MB (even though it was using only a couple hundred MBs).

Conversely, if we were to allocate too little memory for a lambda function, it would simply fail with an out-of-memory error, so we won't see any of our lambda functions hitting the 3008MB mark, otherwise they would have already been failing at the lower memory settings. Besides 3008MB is the largest setting.

My point using 3008MB is that someone has already done the work of figuring this out. Of course, I'm not taking the author's word as gospel, but rather simply using his findings as a likely better starting point than the out-of-box Cumulus settings.

Of course, that doesn't mean it's not worth revisiting this to do some of our own verification/confirmation of how well the author's findings fit our scenario, so I'm not ruling out revisiting this, and I appreciate your flagging this as something to revisit. I'm just using settings that are likely at least better than the Cumulus defaults.

Comment (krisstanton) [reply app/stacks/cumulus/main.tf:1]: Sorry, I missed the detail about 3008 MB being the sweet spot and I also didn't realize 3008 MB was the max setting.  
Thanks for clarifying all of that and adding the detail about the automatic vCPU allocation and the article.

Looking forward to seeing the performance gains!
